File: Server/src/main/kotlin/server/Command.kt
package server

interface Command


/**
 *  Commands sent by the client and received by the [Server]
 */
enum class ReceiveCommand: Command {
    MOVE, INTERACT_WITH_FOOD_BOX, CONNECTION_REQUEST, DISCONNECTED, STALE, GAME_STARTED
}

/**
 *  Commands sent by the [Server] to the clients
 */
enum class SendCommand: Command {
    CONNECTION_DENIED, CONNECTION_ACCEPTED, START_GAME, GAME_ABORTED, GAME_OVER, UPDATE_STATE
}

/**
 *  Corresponds to a direction that a client wants to move its corresponding [game.Player]
 */
enum class Direction {
    UP, DOWN, LEFT, RIGHT
}

File: Client/_communication/command.py
from enum import Enum


class SendCommand(str, Enum):
    """
        Commands that are sent from this client to the server.
    """
    MOVE = "MOVE"
    INTERACT_WITH_FOOD_BOX = "INTERACT_WITH_FOOD_BOX"
    CONNECTION_REQUEST = "CONNECTION_REQUEST"
    DISCONNECTED = "DISCONNECTED"
    GAME_STARTED = "GAME_STARTED"


class ReceiveCommand(str, Enum):
    """
        Commands that are sent from the server to this client.
    """
    CONNECTION_DENIED = "CONNECTION_DENIED"
    CONNECTION_ACCEPTED = "CONNECTION_ACCEPTED"
    START_GAME = "START_GAME"
    GAME_ABORTED = "GAME_ABORTED"
    GAME_OVER = "GAME_OVER"
    UPDATE_STATE = "UPDATE_STATE"
